VocaDbWeb/New/pages/discussion/folders/[id].tsx

```diff
@@ -1,7 +1,148 @@
-import { useRouter } from 'next/router';
+import { GetServerSideProps, InferGetServerSidePropsType } from 'next';
+import { DiscussionFolderContract } from '@/types/DataContracts/Discussion/DiscussionFolderContract';
+import { DiscussionTopicContract } from '@/types/DataContracts/Discussion/DiscussionTopicContract';
+import { apiFetch } from '@/Helpers/FetchApiHelper';
+import useStyles from './[id].styles';
+import { IconBook2 } from '@tabler/icons-react';
+import { Text, Box, ThemeIcon, Pagination, Table, useMantineTheme } from '@mantine/core';
+import Link from 'next/link';
+import CustomImage from '@/components/Image/Image';
 
-export default function FolderPage() {
-	const router = useRouter();
-	return <p>Folder: {router.query.id}</p>;
+export default function FolderPage({
+	folder,
+	start,
+	page,
+	totalPage,
+	topics,
+}: InferGetServerSidePropsType<typeof getServerSideProps>) {
+	const styles = useStyles();
+	const setPage = (page: number) => {
+		window.location.href = `/discussion/folders/${folder.id}?page=${page}`;
```

Comment:
Please use the nextjs [useRouter ](https://nextjs.org/docs/pages/api-reference/functions/use-router) api